### DISCUSSION TITLE
Difficulty Setting up AutoDRIVE and establsihing ROS connection

### DISCUSSION THREAD

**Discussion by SumeetShanbhag** [Tinker-Twins/AutoDRIVE#5] [Q&A] [answered]
I had some difficulty setting up when I was trying to use the AutoDRIVE simulator and DevKit that you had uploaded on GitHub.  While I was able to run the Simulator, I had a few issues with the following.
1.	I was not able to establish a connection using IP address and port number in AutoDrive Simulator and thus was not able to run in Autonomous mode. 
2.	I was not able to run the launch files in ADSS.
3.	I also tried to use the ROS packages but was not able to run the launch files there without establishing connection. 
 
I tried using the downgraded version of python 3.6.5 as mentioned in the dependencies and also installed all required dependencies again but it still doesn't work for me. I am using a conda environment currently.

**Comment by Tinker-Twins** [comment, marked as answer]
Hope our discussion via video call was helpful! Here is a quick review of the discussion:

Following is a list of dependencies for using [AutoDRIVE Simulator](https://github.com/Tinker-Twins/AutoDRIVE/tree/AutoDRIVE-Simulator) (tested with [AutoDRIVE Simulator 0.2.0](https://github.com/Tinker-Twins/AutoDRIVE/releases/tag/Simulator-0.2.0)) with ROS (tested with [ROS Noetic](http://wiki.ros.org/noetic)):

- Websocket-related dependencies for communication bridge between [AutoDRIVE Simulator](https://github.com/Tinker-Twins/AutoDRIVE/tree/AutoDRIVE-Simulator) and [AutoDRIVE Devkit](https://github.com/Tinker-Twins/AutoDRIVE/tree/AutoDRIVE-Devkit) (version sensitive):
  | Package | Tested Version |
  |---------|----------------|
  | eventlet | 0.33.3 |
  | Flask | 1.1.1 |
  | Flask-SocketIO | 4.1.0 |
  | python-socketio | 4.2.0 |
  | python-engineio | 3.13.0 |
  | greenlet | 1.0.0 |
  | gevent | 21.1.2 |
  | gevent-websocket | 0.10.1 |
  
  ```bash
  $ pip3 install eventlet==0.33.3
  $ pip3 install Flask==1.1.1
  $ pip3 install Flask-SocketIO==4.1.0
  $ pip3 install python-socketio==4.2.0
  $ pip3 install python-engineio==3.13.0
  $ pip3 install greenlet==1.0.0
  $ pip3 install gevent==21.1.2
  $ pip3 install gevent-websocket==0.10.1
  ```

- Generic dependencies for data processing and visualization (usually any version will do the job):

  | Package | Tested Version |
  |---------|----------------|
  | attrdict | 2.0.1 |
  | numpy | 1.13.3 |
  | pillow | 5.1.0 |
  | opencv-contrib-python | 4.5.1.48 |

  ```bash
  $ pip3 install attrdict
  $ pip3 install numpy
  $ pip3 install pillow
  $ pip3 install opencv-contrib-python
  ```

We were able to reproduce the errors you were facing and (more importantly) resolve them. We have updated the repository with [detailed documentation](https://github.com/Tinker-Twins/AutoDRIVE/blob/AutoDRIVE-Devkit/ADSS%20Toolkit/autodrive_ros/README.md) to help you (and others) with dependencies, setup and usage. Kindly check out and clone the [updated AutoDRIVE Devkit branch](https://github.com/Tinker-Twins/AutoDRIVE/tree/AutoDRIVE-Devkit/ADSS%20Toolkit/autodrive_ros). Furthermore, we have released [AutoDRIVE Devkit 0.2.1](https://github.com/Tinker-Twins/AutoDRIVE/releases/tag/Devkit-0.2.1) with the updated software and instructions as a stable benchmark for convenience.

It seems like some of your packages might have conflicting versions/dependencies. Can you try uninstalling all the packages listed as [AutoDRIVE Devkit's ROS Package](https://github.com/Tinker-Twins/AutoDRIVE/tree/AutoDRIVE-Devkit/ADSS%20Toolkit/autodrive_ros) dependencies (even the ones that might be installed using conda, apt, etc.) and then installing the specific versions mentioned above using pip3 (and NOT using conda, apt, etc.)? Also, try deactivating any active conda environments before execution.

Hope this is helpful to you (as well as others looking in this direction in the future)!

**Reply by SumeetShanbhag** [reply]
Hello guys, 

Thank you so much for helping me out. It turns out that the issue was with the conda environment and I was finally able to establish communication with ros_bridge as well. Thank you so much for reaching out and updating the repositories
